File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int main(void)
{
    // t = text     w = words       l = letters
    string t = get_string("Text: ");
    int w = 1;
    int l;
    int lc = 0;
    int s = 0;

    for (l = 0; t[l] != '\0'; l++)
    {
        if (t[l] == ' ')
        {
            w++;
        }

        if (isalpha(t[l]) != 0)
        {
            lc++;
        }
        if (t[l] == '.' || t[l] == '?' || t[l] == '!')
        {
            s++;
        }

    }
    float index = 0.0588 * lc / w * 100 - 0.296 * s / w * 100 - 15.8;
    int indexr = round(index);
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", indexr);
    }

}

File: pset6/Readability/readability.py
from cs50 import get_string

def main():
    while True:
        text = get_string("Text: ")
        if len(text) > 0:
            break

    lettercount = 0
    wordcount = 1
    sentencecount = 0

    for i in text:
        if i.isalpha() == True:
            lettercount += 1
        if i in ['.', '!', '?']:
            sentencecount += 1
        if i == ' ':
            wordcount += 1

    index = 0.0588 * lettercount * 100 / wordcount - 0.296 * sentencecount * 100 / wordcount - 15.8

    if index < 16 and index > 1:
        print(f"Grade {round(index)}")

    elif index >= 16:
        print(f"Grade 16+")

    elif index <= 1:
        print("Before Grade 1")




main()